### PR TITLE
drivers/ieee802154: Fix KW41Z driver fault

### PR DESCRIPTION
**Pull request by scpf19**
KW41Z IEEE802154 driver fix, by adding an empty configure function
to resolve HardFault at start up

Signed-off-by: Tobias Schlaepfer <scpf@zhaw.ch>

### CODE REVIEW

**Comment by rlubos** [comment]
@scpf19 You need to fix the commit message before we can approve and merge, see the "GItlint issues" section. For the signoff issue, you likely need to configure a proper user name in your git, see https://help.github.com/en/articles/setting-your-username-in-git.

File: drivers/ieee802154/ieee802154_kw41z.c [hunk 1]
```diff
@@ -1020,6 +1020,12 @@ static void kw41z_iface_init(struct net_if *iface)
 	ieee802154_init(iface);
 }
 
+static int kw41z_configure(struct device *dev, enum ieee802154_config_type type,
+		   const struct ieee802154_config *config)
```
Comment (tbursztyka) [inline drivers/ieee802154/ieee802154_kw41z.c]: indent the parameter below last '(' + 1 space

File: drivers/ieee802154/ieee802154_kw41z.c [hunk 2]
```diff
@@ -1020,6 +1020,12 @@ static void kw41z_iface_init(struct net_if *iface)
...
+{
+	return 0;
```
Comment (bwitherspoon) [inline drivers/ieee802154/ieee802154_kw41z.c]: This should return a negative errno if the configure type is not supported.

**Comment by tbursztyka** [comment]
Merge you 3 commits into 1

**Review by jukkar — APPROVED** [review]
LGTM

**Comment by dleach02** [comment]
I approve the changes. 

But as a side note, I find it slightly annoying that we add items to the configuration block knowing full well that if we don't provide an error check from the calling side on implementation of the function that it will cause a fault. In the future, if we expand the driver API (for any driver), we need to add "issues" to update the drivers using the API (at a minimum) or go in and change all the drivers to add the support.

**Review by dleach02 — APPROVED** [review]

**Pull request merged**